src/adapter/src/catalog/storage.rs

```diff
@@ -526,6 +526,38 @@ async fn migrate(
         |_, _, _| Ok(()),
         |_, _, _| Ok(()),
         |_, _, _| Ok(()),
+        // Add USAGE privileges on the mz_system cluster for the mz_introspection user.
+        //
+        // Introduced in v0.55.0
+        //
+        // TODO(jkosh44) Can be cleared (patched to be empty) in v0.58.0
+        |txn: &mut Transaction<'_>, _now, _bootstrap_args| {
+            txn.clusters.update(|_cluster_key, cluster_value| {
+                let mz_introspection_privilege = MzAclItem {
+                    grantee: MZ_INTROSPECTION_ROLE_ID,
+                    grantor: MZ_SYSTEM_ROLE_ID,
+                    acl_mode: AclMode::USAGE,
+                };
+                if &cluster_value.name == MZ_SYSTEM_CLUSTER.name
+                    && !cluster_value
+                        .privileges
+                        .as_ref()
+                        .expect("cluster privileges not migrated")
```

Comment:
nit: instead of using `.expect(...)` here, can you do something like:
```
.map(|priv| priv.contains(...)
.unwrap_or(false)
```

Reply:
Done